src/mongocxx/collection.cpp

```diff
@@ -98,11 +98,15 @@ cursor collection::find(bsoncxx::document::view filter, const options::find& opt
     scoped_bson_t filter_bson;
     scoped_bson_t projection(options.projection());
 
-    if (options.modifiers()) {
-        filter_builder << "$query" << bsoncxx::types::b_document{filter}
-                       << builder::stream::concatenate{
-                              options.modifiers().value_or(document::view{})};
-
+    if (options.sort() || options.modifiers()) {
+        filter_builder << "$query" << bsoncxx::types::b_document{filter};
```

Comment:
Is there a reason we don't want to use $query when we don't have any modifiers?  I'm wondering if we can simplify the logic here.